bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/ZkLedgerUnderreplicationManager.java

```diff
@@ -884,4 +885,39 @@ public String getReplicationWorkerIdRereplicatingLedger(long ledgerId)
         }
         return replicationWorkerId;
     }
+
+    @Override
+    public void setCheckAllLedgersCTime(long checkAllLedgersCTime) throws UnavailableException {
+        LOG.debug("setCheckAllLedgersCTime");
+        try {
+            if (zkc.exists(checkAllLedgersCtimeZnode, false) != null) {
+                zkc.setData(checkAllLedgersCtimeZnode, Long.toString(checkAllLedgersCTime).getBytes(UTF_8), -1);
```

Comment:
can you make a util function for `Long.toString(checkAllLedgersCTime).getBytes(UTF_8)`? I see it is used multiple places.

regardless, shall we consider using a protobuf structure for future extensibility?

Reply:
added protobuf